libfdt/meson.build

```diff
@@ -26,27 +26,33 @@ else
 endif
 
 link_args += version_script
-libfdt = both_libraries(
+libfdt = library(
   'fdt', sources,
   version: meson.project_version(),
   link_args: link_args,
   link_depends: 'version.lds',
   install: true,
 )
 
-if static_build
-  link_with = libfdt.get_static_lib()
-else
-  link_with = libfdt.get_shared_lib()
-endif
-
 libfdt_inc = include_directories('.')
 
 libfdt_dep = declare_dependency(
   include_directories: libfdt_inc,
-  link_with: link_with,
+  link_with: libfdt,
 )
 
+if static_build
+  libfdt_static_dep = declare_dependency(
+    include_directories: libfdt_inc,
+  )
+  libfdt_static_objs = libfdt.extract_all_objects()
```

Comment:
IIUC, this is doing static linking by individually linking against every constituent `.o` file.  I guess that will work, but it seems convoluted, and I think we still want the ability to build a `.a` static library.

Reply:
I wasn't happy with this approach either, it was the only way I could think to allow for `static-build=true` to require static linking, while also not requiring Meson compiles a static archive when `static-build=false`.

Switching from `both_libraries()` to `library()` doesn't stop us from building a `.a` static archive. It just changes it so Meson won't always build it, but instead build one or the other depending on the build configuration.

Reply:
Looking over this with fresh eyes, I think a much easier solution would be to switch between `library()` and `static_library()`. I pushed an update with that change.